### PR TITLE
[GPU] Add BF16 support to rms kernels

### PR DESCRIPTION
**Pull request by clee30**
Enable BF16 for the RMS OCL implementation and the ref/opt kernels. BF16 data is decoded into the FP32 compute domain (via the standard DECODE_INPUT0/1_COMPUTE_TYPE mechanism) so the RMS reduction, scaling, and epsilon are computed in FP32 rather than on raw BF16 bit patterns.

Add BF16 unit tests for both kernels, including a near-zero case that guards the epsilon term in the rsqrt.

### Tickets:
 - [CVS-192281](https://jira.devtools.intel.com/browse/CVS-192281)

### AI Assistance:
 - *AI assistance used: no / yes*
 - *If yes, summarize how AI was used and what human validation was performed (build/tests/manual checks).*

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl [hunk 1]
```diff
@@ -50,28 +50,28 @@ KERNEL(rms_gpu_ref)(
...
                 const uint gamma_idx = INPUT1_OFFSET + (INPUT1_LENGTH == 1 ? 0 : n);
-                OUTPUT_TYPE result = TO_OUTPUT_TYPE(rms) * TO_OUTPUT_TYPE(input[input_idx]) * TO_OUTPUT_TYPE(gamma[gamma_idx]);
+                ACCUMULATOR_TYPE result = rms * DECODE_INPUT0_COMPUTE_TYPE(input[input_idx]) * DECODE_INPUT1_COMPUTE_TYPE(gamma[gamma_idx]);
 #else
-                OUTPUT_TYPE result = TO_OUTPUT_TYPE(rms) * TO_OUTPUT_TYPE(input[input_idx]);
+                ACCUMULATOR_TYPE result = rms * DECODE_INPUT0_COMPUTE_TYPE(input[input_idx]);
 #endif
                 #if HAS_FUSED_OPS
                     FUSED_OPS;
                     result = FUSED_OPS_RESULT;
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl:73]: `result` dtype changed to `ACCUMULATOR_TYPE`, but `FUSED_OPS` still use output type
Mismatches can lead to problems, please fix and add some bf16 fusing tests for RMS

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl [hunk 2]
```diff
@@ -42,20 +42,6 @@
 #define HAS_FUSED_OPS_DECLS 0
 #endif
 
-#if SUBGROUP_BLOCK_SIZE == 1
```
Comment (Lyamin-Roman) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl:12]: If this is really not used, then we can remove the jit constant definition and its calculation from `rms_kernel_bfyx_opt.cpp`

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl [hunk 3]
```diff
@@ -50,28 +50,28 @@ KERNEL(rms_gpu_ref)(
...
                 #endif
-                output[output_idx] = result;
+                output[output_idx] = TO_OUTPUT_TYPE(result);
```
[the remark this answers is not in the document]
Comment (clee30) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl]: resolved

File: src/plugins/intel_gpu/tests/unit/test_cases/rms_gpu_test.cpp [hunk 4]
```diff
@@ -801,3 +801,118 @@ TEST(rms_gpu_test, in_place_crop_rms_spatial_split) {
...
+
+    auto input = engine.allocate_memory({in_shape, data_types::bf16, format::bfyx});
+    // Rank-3 input [1, C, W] normalizes over the last dim W; gamma covers W elements.
+    const size_t W = in_shape[in_shape.size() - 1].get_length();
+    memory::ptr gamma = nullptr;
+    if (with_gamma)
+        gamma = engine.allocate_memory({ov::PartialShape{1, static_cast<int64_t>(W)}, data_types::bf16, format::bfyx});
+    auto output_ref = engine.allocate_memory({in_shape, data_types::bf16, format::bfyx});
+
+    tests::set_random_values<ov::bfloat16>(input, true, in_min, in_max);
```
[the remark this answers is not in the document]
Comment (clee30) [reply src/plugins/intel_gpu/tests/unit/test_cases/rms_gpu_test.cpp]: fixed

File: src/plugins/intel_gpu/tests/unit/test_cases/rms_gpu_test.cpp [hunk 5]
```diff
@@ -801,3 +801,118 @@ TEST(rms_gpu_test, in_place_crop_rms_spatial_split) {
...
+        float diff = std::abs(gpu_val - ref_val);
+        float tolerance = std::max(abs_floor, std::abs(ref_val) * rel_tol);
+        ASSERT_LE(diff, tolerance) << "Mismatch at i=" << i
+            << " gpu=" << gpu_val << " ref=" << ref_val << " diff=" << diff;
+    }
+}
+
+// BF16 RMS on the ref kernel, small rank-3 shape, with gamma
+TEST(rms_gpu_test, rms_test_bf16_bfyx_ref) {
+    run_rms_bf16("rms_gpu_ref", ov::PartialShape{1, 2, 6}, /*with_gamma=*/true, /*dynamic=*/false);
```
[the remark this answers is not in the document]
Comment (clee30) [reply src/plugins/intel_gpu/tests/unit/test_cases/rms_gpu_test.cpp:888]: added

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl (hunk 2 above)
Comment (clee30) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_bfyx_opt.cl:12]: removed.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl (hunk 1 above)
Comment (clee30) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/rms_gpu_ref.cl:73]: Updated.  Please check

**Review by e-ddykim — APPROVED** [review]
Looks good to me

**Review by Lyamin-Roman — APPROVED** [review]
